scripts/enrich_rulings_ai.py

```
@@ -0,0 +1,276 @@
+
+    # Initial Provenance from external source structure
+    provenance = {
+        "source_type": external_ruling.get("source_type_hint", "unknown_external"),
+        "source_name": None, # To be filled by AI or manual review
+        "source_date": None, # To be filled by AI or manual review
+        "retrieval_date": external_ruling.get("retrieval_date_utc", datetime.datetime.utcnow().isoformat()),
+        "source_url": external_ruling.get("source_url_or_context")
+    }
+
+    # Attempt to extract more details from text using AI placeholder
+    provenance = ai_extract_provenance_details(raw_text, provenance)
```

Ruff (0.11.9)

`datetime.datetime.utcnow()` used

(DTZ003)